packages/js/components/src/experimental-select-control/select-control.tsx

```diff
@@ -103,20 +114,51 @@ export const SelectControl = ( {
 
 					break;
 				case useCombobox.stateChangeTypes.InputKeyDownEnter:
+				case useCombobox.stateChangeTypes.FunctionSelectItem:
 				case useCombobox.stateChangeTypes.ItemClick:
 				case useCombobox.stateChangeTypes.InputBlur:
 					if ( selectedItem ) {
 						onSelect( selectedItem );
-						setInputValue(
-							multiple ? '' : itemToString( selectedItem )
-						);
+						if ( clearSearchOnSelect ) {
+							onInputChange( '' );
+							setInputValue( '' );
+						} else if ( ! multiple ) {
+							onInputChange( itemToString( selectedItem ) );
+							setInputValue( itemToString( selectedItem ) );
+						}
 					}
 
 					break;
 				default:
 					break;
 			}
 		},
+		stateReducer: ( state, actionAndChanges ) => {
+			const { changes, type } = actionAndChanges;
+
+			switch ( type ) {
+				case useCombobox.stateChangeTypes.InputKeyDownEnter:
+				case useCombobox.stateChangeTypes.FunctionSelectItem:
+				case useCombobox.stateChangeTypes.ItemClick:
+					return {
+						...changes,
+						isOpen: keepMenuOpenOnSelect ? true : false,
```

Comment:
Is the addition of `stateReducer` mostly for the sake of this prop change?  If so, in line with my earlier comment, I think the consumer could handle controlling special open state or rendering.

There's some level of prior art with this as `isOpen` can be passed to the child for menu render.

Reply:
Going this route felled like a more native approach to `downshift`. If I wanted to handle the `isOpen` manually and pass it to the menu, I would have to keep a create a separate state outside of the `SelectControl` component to toggle this, and make sure it adheres to all the above events as well.
This would mean that we probably want to expose a callback to the `stateReducer` change anyhow.

To me this felled overkill, and adding an option for this here felled better. I was following this example of Downshift -> https://www.downshift-js.com/use-combobox/#basic-multiple-selection with leaving the dropdown open.

This feature does feel like it could be more popular. I could still expose a callback for the state reducer changes, if that seems of benefit?
Or would you prefer to expose the entire `stateReducer` function? Although this would require the user to have more knowledge of `downshift`.